src/main/java/org/jooq/lambda/SeqBuffer.java

```diff
@@ -57,6 +63,17 @@ private SeqBuffer(Spliterator<T> source) {
         return Seq.seq(new BufferSpliterator());
     }
 
+    private static <T> List<T> initBuffer(Spliterator<T> source) {
+        long estimateSize = source.estimateSize();
+        if (estimateSize == Long.MAX_VALUE) // infinite/unknown size
+            return new ArrayList<>();
+
+        if (estimateSize > MAX_ARRAY_SIZE)
+            throw new IllegalArgumentException("Stream is too long to be buffered: " + estimateSize);
+
+        return new ArrayList<>((int) estimateSize);
```

Comment:
Hm, no I don't agree with this. If we create a large capacitied array list, we're back to the original solution that you wanted to avoid: Up-front large memory consumption for intermediate buffers. Imagine:

```java
Seq.seq(Collections.nCopies(1000000, "value")).splitAt(999999).v1.limit(1).forEach(System.out::println);
```

The `splitAt()` operation would create a large array list only to consume its first element...